sdk/cognitivelanguage/azure-ai-language-conversations/samples/authoring/sample_assign_deployment_resources.py

```diff
@@ -0,0 +1,94 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+FILE: sample_assign_deployment_resources.py
+
+DESCRIPTION:
+    This sample demonstrates how to assign deployment resources to a project and deploy a project to specific deployment resources.
+
+USAGE:
+    python sample_assign_deployment_resources.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_CONVERSATIONS_ENDPOINT             - endpoint for your CLU resource.
+    2) AZURE_CLIENT_ID                          - the client ID of your active directory application.
+    3) AZURE_TENANT_ID                          - the tenant ID of your active directory application.
+    4) AZURE_CLIENT_SECRET                      - the secret of your active directory application.
+"""
+
+"""
+Assigning deployment resources requires you to authenticate with AAD, you cannot assign deployment resources with key based authentication. 
+"""
+def sample_assign_resources():
+    import os
+    from azure.identity import DefaultAzureCredential
+    from azure.ai.language.conversations.authoring import ConversationAuthoringClient
+
+    clu_endpoint = os.environ["AZURE_CONVERSATIONS_ENDPOINT"]
+    
+    client = ConversationAuthoringClient(
+        clu_endpoint, credential=credential
+    )
+    
+    project_name = "test_project"
+
+    credential = DefaultAzureCredential()
+
+    poller = client.begin_assign_resource(
+        project_name=project_name,
+        resourcesMetadata: [
+          {
+            "azureResourceId": "/subscriptions/80000000-0000-4e8a-0000-7ce285849735/resourceGroups/test-rg/providers/Microsoft.CognitiveServices/accounts/LangTestWeu",
+            "customDomain": "lang-test-weu.cognitiveservices.azure.com",
+            "region": "westeurope"
```

Comment:
These values look like they should be included as environment variables in the sample

Reply:
Nope these wouldn't be environment variables, they're the payload the customer provides - they do happen to look like env variables

Reply:
If they aren't passed as env vars, we won't be able to test run this sample in CI to ensure that our sample code works. I do see the value in showing the "shape" of these of parameter values, though, maybe we can include it in a comment?

Reply:
I'd keep it in, because we'd still need valid values for CI, do all the samples have to be automatically ran? This one was always going to be complicated anyways, it requires AAD authentication with specific roles assigned to both the source and target resource.

Reply:
Discussed offline, we'll skip running this sample for now. To skip, please add the sample file names to this list here:

https://github.com/Azure/azure-sdk-for-python/blob/main/scripts/devops_tasks/test_run_samples.py#L141

Should look like:

```python
    "azure-ai-language-conversations": [
        "sample_assign_deployment_resources.py",
        "sample_assign_deployment_resources_async.py",
    ],
```